File: SmartHome/app/ingternal/room/schemas/room.py
from typing import List, Optional, Dict, Tuple
from pydantic import BaseModel
from app.ingternal.room.schemas.type_device import DeviceTypeModel

class RoomCreate(BaseModel):
    name_room: str

class DeviceRoom(BaseModel):
    system_name: str
    poz: Optional[str] = None

class RoomDevicesUpdate(BaseModel):
    devices: List[DeviceRoom]

class RoomUpdate(BaseModel):
    name_room: str

class RoomDevicesRaw(BaseModel):
    name_room: str
    devices: List[DeviceRoom]
    device_room: Optional[Dict[str, DeviceTypeModel]] = None

class RoomDevciesRawList(BaseModel):
    rooms: List[RoomDevicesRaw]

class RoomDevicesLink(BaseModel):
    name_room: str
    device: str

class RoomDevicesMove(BaseModel):
    name_room: str
    device: str
    poz: Optional[str] = None

class RoomDevicesSet(BaseModel):
    device_type: str
    field_name: str
    value: str
    name_room: str

File: frontend/client_base/src/entites/rooms/models/room.ts

export interface RoomDevice {
    system_name: string
    poz?: string
}

export interface DeviceField {
  system_name: string;
  id_field_device: string;
}

export interface DeviceFieldType {
  field_type: string;
  devices: DeviceField[];
}

export interface DeviceTypeModel {
  // ключ — это name_field_type
  fields: Record<string, DeviceFieldType>;
}

export interface Room {
    name_room: string,
    devices: RoomDevice[]
    device_room: Record<string, DeviceTypeModel>
}

export interface Rooms {
    rooms: Room[]
}

export interface RoomCreate {
    name_room: string
}

export interface RoomDevices {
    devices: RoomDevice[]
}